### ISSUE TITLE
Add race method?

### ISSUE THREAD

**Issue by ivan-kleshnin** [WebReflection/dodgy#1]
Analog of `Promise.race` which would cancel all non-fulfilled `Dodges` (or whatever you call it).

**Comment by WebReflection** [comment]
I guess `Promise.all` would be welcome too, right?

**Comment by ivan-kleshnin** [comment]
Yep.

**Comment by WebReflection** [comment]
I will try to find some time to implement both soon

**Comment by WebReflection** [comment]
How should we spec `.race` ?
- first one that completes cancels all others
- a `Dodgy.race` can be canceled regardless anyway

Would this work?

And how should we spec `.all` ?
- if any in the list get canceled the whole thing cancels too ?
- a `Dodgy.all` can be canceled but any of its promises might have resolved already ?

I think `race` has less surprises than `.all` ... maybe I'll do that first ... also because `.all` usually doesn't make sense combined with cancelability, IMO

What do you think?

**Comment by ivan-kleshnin** [comment]
I will substitute `Dodgy` with `Task` cause first term makes no sense to my brain, sorry.

So I would specify desired behavior like this. 

As long as `Task` is resolved it can't be canceled. Just as `Promise`.

`Task.race([task1, task2])` returns `Task` which can be canceled.
If this **outcoming** `Task` is canceled – every **incoming** task is canceled. But not vice versa!
Canceling of any incoming `Task` cancels just it. Unless it was the only incoming task.
In that case everything is canceled. This looks totally predictable.

`all` case is more tricky, as you noted. But I would repeat the same behavior here.
Canceling incoming task does not affect outcoming task. Unless it was the single item.
In that case everything is canceled.

**Comment by WebReflection** [comment]
I think version `1.2.0` with tests I've just pushed should work fine.

Described in here: https://github.com/WebReflection/dodgy#dodgyraceiterable

TL;DR if you abort the race it aborts all abortable and as soon as there is a result any other cancelable will be canceled.

I wouldn't go trickier than this, let me know if this works for you. Thanks.

**Comment by ivan-kleshnin** [comment]
Sure, thanks! I'll check it out and respond in a while.

**Comment by ivan-kleshnin** [comment]
Works as expected, thank you again.
